tm2/pkg/p2p/transport.go

```diff
@@ -200,18 +196,24 @@ func (mt *MultiplexTransport) runAcceptLoop() {
 
 	for {
 		select {
-		case <-mt.ctx.Done():
+		case <-ctx.Done():
 			mt.logger.Debug("transport accept context closed")
-
 			return
+
 		default:
 			// Accept an incoming peer connection
 			c, err := mt.listener.Accept()
 			if err != nil {
-				mt.logger.Error(
-					"unable to accept p2p connection",
-					"err", err,
-				)
+				switch {
+				case goerrors.Is(err, context.Canceled), goerrors.Is(err, context.DeadlineExceeded):
+					// Context has been canceled or deadline exceeded, exit loop
```

Comment:
same as above.

Reply:
Updated, the comment should better match the behaviors.